src/subscriptions/model.rs

```diff
@@ -78,6 +82,35 @@ impl Subscriptions {
     }
 }
 
+impl Subscription {
+    pub async fn save<'a, E>(&self, executor: E) -> Result<(), SubscriptionsError>
+    where
+        E: Executor<'a>,
+    {
+        let mut transaction = executor.begin().await?;
+        sqlx::query!(
+            r#"
+            INSERT INTO subscription (uuid_id, user_id, notify_mailman, date)
+                SELECT ?, ?, ?, ?
+                WHERE NOT EXISTS
+                (SELECT id FROM subscription WHERE uuid_id = ? AND user_id = ?)
```

Comment:
This doesn't have the correct behavior yet. If the subscription already exists, we should still update `notify_mailman`.